src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,149 @@
+import React, {
+  RefObject,
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react';
+import './Todo.scss';
+import { ActionInTodoContextReducer, TodoInterface } from '../../utils';
+
+interface Props {
+  readonly todos: TodoInterface[];
+  updateTodo: React.Dispatch<ActionInTodoContextReducer>;
+  mainInput: RefObject<HTMLInputElement>;
+}
+
+export const Todo = React.memo<Props>(({ todos, updateTodo, mainInput }) => {
+  const allData = useMemo(
+    () =>
+      todos.map(e => {
+        return { ...e };
+      }),
+    [todos],
+  );
+  const [listClickedsOrNot, setListClickedsOrNot] = useState(
+    allData.map(() => false),
+  );
+  const [, forceRender] = useState({});
+
+  useEffect(() => {
+    setListClickedsOrNot(allData.map(() => false));
+  }, [allData]);
+
+  const handleClickedOrNot = (idx: number) => {
```

Comment:
always use useCallback on handlers (nice article about it) -> https://semaphoreci.com/blog/react-usecallback-hook#:~:text=useCallback%20is%20most%20commonly%20used,as%20event%20handlers%20and%20timers.

Reply:
Hi @jv-aquino, the many warnings is about the use of operator ! (assertation not null operator). I made flexible objetcts and types, if I trying to stop this warnings I'll have a much more verbose code, I guess.

About useCallback, in fact some handlers have assumption of closures and useCallback can made some unspected behavior, but I think that in fact exists handlers that useCallback can avoid unecessary 'realocation' memory btw renderizations. 

Again thank you for advices, I'll proceed with the corrections. :-D

Reply:
ohh sorry, I just saw a lot of them and thought there was some problems!